From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 19 Feb 2026 04:32:21 +0100
Subject: document-portal: Do not try to use alternate method for file access
 for snaps

It's all been designed around flatpak internals, so it won't ever work
with snaps.

Just return early if snap routine file-access fails

Origin: https://github.com/flatpak/xdg-desktop-portal/pull/1764
---
 document-portal/document-portal.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/document-portal/document-portal.c b/document-portal/document-portal.c
index 3121ace..a2c148d 100644
--- a/document-portal/document-portal.c
+++ b/document-portal/document-portal.c
@@ -649,6 +649,12 @@ app_has_file_access (const char *target_permissions_id,
     {
       res = xdp_spawn (&error, "snap", "routine", "file-access",
                         target_permissions_id + strlen ("snap."), path, NULL);
+
+      if (!res)
+        {
+          /* Fallback access method has no support for snaps. */
+          return FALSE;
+        }
     }
   else
     {
